src/main/java/com/google/devtools/build/lib/remote/ByteStreamBuildEventArtifactUploader.java

```diff
@@ -174,13 +179,19 @@ public boolean isRemote() {
     public boolean isOmitted() {
       return omitted;
     }
+
+    public DigestFunction.Value getDigestFunction() {
+      return digestFunction;
+    }
   }
 
   /**
    * Collects metadata for {@code file}. Depending on the underlying filesystem used this method
    * might do I/O.
    */
   private PathMetadata readPathMetadata(Path path, LocalFile file) throws IOException {
+    DigestUtil digestUtil = new DigestUtil(xattrProvider, path.getFileSystem().getDigestFunction());
```

Comment:
I think we can just use `path.getFileSystem().getDigestFunction()` instead of creating a `DigestUtil`.

Reply:
DigestUtil exposes the DigestFunction.Value which is distinct from DigestHashFunction which is returned by the method you're suggesting.

So we're using DigestUtil to convert basically.

Reply:
Also -- sorry it wasn't clear, DigestUtil was already being created in this function (L217), i just moved it up so I can reference it in more places :)

Reply:
Argh, yes. Sorry I didn't read the code carefully.